tests/components/script/test_init.py

```diff
@@ -790,3 +791,35 @@ async def test_script_restore_last_triggered(hass: HomeAssistant) -> None:
     state = hass.states.get("script.last_triggered")
     assert state
     assert state.attributes["last_triggered"] == time
+
+
+@pytest.mark.parametrize("script_mode", SCRIPT_MODE_CHOICES)
+async def test_recursive_script(hass, script_mode):
+    """Test recursive script calls does not deadlock."""
+    assert await async_setup_component(
+        hass,
+        "script",
+        {
+            "script": {
+                "script1": {
+                    "mode": script_mode,
+                    "sequence": [
+                        {"service": "script.script1"},
+                        {"service": "test.script"},
+                    ],
```

Comment:
Maybe we should test other more scenarios, like a script calling another script which in turn calls the first script?

Reply:
yes

Reply:
Technically, that could be a valid scenario though...

Reply:
Should we take the variables/fields passed into the script into account?

Reply:
> that could be a valid scenario though

no, a call script.a -> script.b -> script.a where script.a mode is "restart" or "queued" is not valid because the script will kill or wait for itself, causing a deadlock

> should we take the variables/fields passed into the script into account

how?